packages/application-shell/src/components/login/login.js

```diff
@@ -1,257 +0,0 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-import { FormattedMessage, injectIntl } from 'react-intl';
-import flowRight from 'lodash/flowRight';
-import { Formik } from 'formik';
-import { connect } from 'react-redux';
-import { actions as sdkActions } from '@commercetools-frontend/sdk';
-import {
-  ContentNotification,
-  PasswordField,
-  PrimaryButton,
-  Spacings,
-  Text,
-  TextField,
-} from '@commercetools-frontend/ui-kit';
-import { LOGOUT_REASONS } from '@commercetools-frontend/constants';
-import * as storage from '@commercetools-frontend/storage';
-import { withApplicationContext } from '@commercetools-frontend/application-shell-connectors';
-import { STORAGE_KEYS } from '../../constants';
-import PublicPageContainer from '../public-page-container';
-import LoginBox from '../login-box';
-import Countdown from './countdown';
-import messages from './messages';
-import { validate } from './validations';
-
-const defaultTargetUrl = window.location.origin;
-
-const redirect = targetUrl => window.location.replace(targetUrl);
-
-export class Login extends React.PureComponent {
-  static displayName = 'Login';
-
-  static propTypes = {
-    location: PropTypes.shape({
-      query: PropTypes.shape({
-        reason: PropTypes.string,
-        redirectTo: PropTypes.string,
-      }).isRequired,
-    }).isRequired,
-    history: PropTypes.shape({
-      push: PropTypes.func.isRequired,
-    }).isRequired,
-
-    // Injected
-    requestAccessToken: PropTypes.func.isRequired,
-    adminCenterUrl: PropTypes.string.isRequired,
-    intl: PropTypes.shape({
-      formatMessage: PropTypes.func.isRequired,
-    }).isRequired,
-  };
-
-  state = {
-    error: this.props.location.query.reason,
-  };
-
-  handleSubmit = (formValues, addHandlers) =>
-    addHandlers(
-      this.props.requestAccessToken({
-        email: formValues.email,
-        password: formValues.password,
-      })
-    )
-      .then(() => {
-        // Set a flag to indicate that the user has been authenticated
-        storage.put(STORAGE_KEYS.IS_AUTHENTICATED, true);
-        // Ensure to remove the IdP Url value from local storage, as in this
-        // case login is not done through SSO.
-        storage.remove(STORAGE_KEYS.LOGIN_STRATEGY);
-        // Redirect to a `redirectTo` url, if present, otherwise to defaul route
-        const nextTargetUrl = this.props.location.query.redirectTo;
-        // Inspect the target url to see if the host is one of commercetools
-        try {
-          const nextTargetUrlObject = new URL(nextTargetUrl);
-          if (nextTargetUrlObject.hostname === window.location.hostname) {
```

Comment:
This is currently preventing the redirect from staging -> localhost (currently a problem only for development).

Reply:
We would change that in the mc-app now though, right? Paring it somehow with `servedByProxy` again I suppose...

Reply:
I guess yes

Reply:
We can tackle that on the PR taking a pre-canary build of this I guess.